vars/checkDockerImage.groovy

```diff
@@ -0,0 +1,39 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+  Checks if the given Docker image exists.
+
+  checkDockerImage(image: 'hello-world:latest')
+*/
+def call(Map args = [:]) {
+  def image = args.containsKey('image') ? args.image : error('checkDockerImage: image parameter is required')
+
+  def redirectStdout = isUnix() ? '2>/dev/null' : '2>NUL'
+  if (cmd(returnStatus: true, script: "docker images -q ${image}  ${redirectStdout}") == 0) {
+    log(level: 'DEBUG', text: "${image} exists in the Docker host")
+    return
+  }
+
+  log(level: 'DEBUG', text: "${image} does not exist: pulling")
+  if (cmd(returnStatus: true, script: "docker pull ${image}") == 0) {
+    return
+  }
```

Comment:
What do you think to add a default new argument to try with pulling?

```
checkDockerImage(image: 'hello-world:latest', pull: true)
```

It can be enabled by default. Then this step can verify the cached docker images or also an image from an external docker registry.

Besides, do you wanna pull or just query if it's available?

Reply:
The bool return is in a upcoming commit!

Reply:
I just want to check. Maybe there is a simpler/faster manner